completers/fastfetch_completer/cmd/root.go

```diff
@@ -0,0 +1,144 @@
+					actionMap[flag.Long] = carapace.ActionExecCommand("fastfetch", "--list-modules", "autocompletion")(func(output []byte) carapace.Action {
+						options := []string{"color", "format"}
+						for _, flags := range groups {
+							for _, flag := range flags {
+								options = append(options, flag.Long)
+							}
+						}
+						for _, line := range strings.Split(strings.TrimRight(string(output), "\n"), "\n") {
+							x, _, _ := strings.Cut(line, ":")
+							options = append(options, x)
+						}
+
+						return carapace.ActionValues(options...)
+					})
+
+				case "config":
+					rootCmd.Flags().StringP(flag.Long, flag.Short, condVal[string](flag.Arg.Default), flag.Desc)
+					actionMap[flag.Long] = carapace.ActionExecCommand("fastfetch", "--list-presets", "autocompletion")(func(output []byte) carapace.Action {
+						presets := strings.Split(strings.TrimRight(string(output), "\n"), "\n")
+						return carapace.ActionValues(presets...)
+					})
+
+				case "enum":
+					rootCmd.Flags().StringP(flag.Long, flag.Short, condVal[string](flag.Arg.Default), flag.Desc)
+					var valuesDescribed []string
+					for key, desc := range flag.Arg.Enum {
+						valuesDescribed = append(valuesDescribed, key, desc)
+					}
+					actionMap[flag.Long] = carapace.ActionValuesDescribed(valuesDescribed...)
+
+				case "logo":
+					rootCmd.Flags().StringP(flag.Long, flag.Short, condVal[string](flag.Arg.Default), flag.Desc)
+					actionMap[flag.Long] = carapace.ActionExecCommand("fastfetch", "--list-logos", "autocompletion")(func(output []byte) carapace.Action {
+						res := []string{"none", "Disable logo", "small", "Show small logo if supported"}
+						for _, logo := range strings.Split(strings.TrimRight(string(output), "\n"), "\n") {
+							res = append(res, logo, "Builtin logo")
+						}
+						return carapace.ActionValuesDescribed(res...)
+					})
+
+				case "num":
+					rootCmd.Flags().IntP(flag.Long, flag.Short, int(condVal[float64](flag.Arg.Default)), flag.Desc)
+
+				case "structure":
+					rootCmd.Flags().StringP(flag.Long, flag.Short, condVal[string](flag.Arg.Default), flag.Desc)
+					actionMap[flag.Long] = carapace.ActionExecCommand("fastfetch", "--list-modules", "autocompletion")(func(output []byte) carapace.Action {
+						var texts []string
+						for _, line := range strings.Split(strings.TrimRight(string(output), "\n"), "\n") {
+							name, desc, _ := strings.Cut(line, ":")
+							texts = append(texts, name, desc)
+						}
+						return carapace.ActionValuesDescribed(texts...).List(":")
+					})
+
+				default:
+					rootCmd.Flags().StringP(flag.Long, flag.Short, condVal[string](flag.Arg.Default), flag.Desc)
+				}
+			} else {
+				rootCmd.Flags().BoolP(flag.Long, flag.Short, false, flag.Desc)
+			}
+		}
+	}
+
+	carapace.Gen(rootCmd).FlagCompletion(actionMap)
+
+	carapace.Gen(rootCmd).PositionalCompletion(
+		carapace.ActionDirectories(),
+	)
+}
+
+func condVal[T any](value interface{}) T {
```

Comment:
Think we can leave this out and just just `false`, `""`, `0`. Unless it is accessed during completion the default values don't really matter.

Reply:
What are the default values used for?

Reply:
Nothing really (unless accessed). It is shown in the help output, which isn't used.

Reply:
However you provides no function overloads that don't require a default value. And we have the value. Why not?

Reply:
That's a requirement of [pflag](https://github.com/spf13/pflag). It adds unnecessary complexity (especially with the use of generics).